gnosis/safe-cmm-app-react
# Feature/#7/market price

# Description

Added `Market Price` component

![screenshot_2020-09-03_13-36-34](https://user-images.githubusercontent.com/43217/92164848-e0ed4e00-edea-11ea-9d0e-9b0605a35a0d.png)

Given both base and quote token addresses as input, component will fetch current price for the pair.

![screenshot_2020-09-03_13-36-45](https://user-images.githubusercontent.com/43217/92164845-e054b780-edea-11ea-8c0c-6a61456e0b3c.png)

- Converted a bunch of components into Pure components
- Refactoring, extracting some new smaller components
- Refactoring, renamed components folders
- Fixed bug on initial token details loading
- Local dev server public address now controlled via envvar `PUBLIC_ENDPOINT`
- Added dev dependency on whydidyourender for local debugging



# To Test
1. Install dependencies: `npm i`
2. Start storybook `npm run storybook`
3. Access http://localhost:6006/?path=/docs/basic-display-marketprice--default

- [ ] Check style matches design
- [ ] Should display stories for `loading`, `no price` and `with price`

# Background

Big PR for apparently simple changes.
Lots of noise due to some folders renamings. 

Tested it on `deploy` page and found issues with hooks, fixed with the help of `whydidyourender`.
Keep in mind deploy page changes are not included.
